src/overlays/actors/ovl_En_Yukabyun/z_en_yukabyun.c

```diff
@@ -27,19 +30,123 @@ const ActorInit En_Yukabyun_InitVars = {
+    func_8002F974(&this->actor, NA_SE_EN_YUKABYUN_FLY - SFX_FLAG);
+}
+
+void func_80B43B6C(EnYukabyun* this, GlobalContext* globalCtx) {
+    this->actor.shape.rot.y += this->unk_150;
+    if (this->actor.xzDistFromLink > 5000.0f) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    func_8002F974(&this->actor, NA_SE_EN_YUKABYUN_FLY - SFX_FLAG);
+}
+
+void func_80B43BCC(EnYukabyun* this, GlobalContext* globalCtx) {
+    func_800297A4(globalCtx, &this->actor.posRot.pos, 8.0f, 0, 0x514, 0x12C, 0xF, 0x5F, 0xA, &D_06000A60);
+    Actor_Kill(&this->actor);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Yukabyun/func_80B43A94.s")
+void EnYukabyun_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnYukabyun* this = THIS;
+    s32 pad;
+    
+    if (((this->collider.base.atFlags & 2) || (this->collider.base.acFlags & 2) ||
+        ((this->collider.base.maskA & 2) && !(this->collider.base.oc->id == ACTOR_EN_YUKABYUN))) ||
+        ((this->actionfunc == func_80B43B6C) && (this->actor.bgCheckFlags & 8)))
+        {
+            this->collider.base.atFlags &= 0xFFFD;
+            this->collider.base.acFlags &= 0xFFFD;
+            this->collider.base.maskA &= 0xFFFD;
+            this->actor.flags &= -6;
+            Audio_PlaySoundAtPosition(globalCtx, &this->actor.posRot.pos, 0x1E, NA_SE_EN_OCTAROCK_ROCK);
+            this->actionfunc = func_80B43BCC;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Yukabyun/func_80B43AD4.s")
+    this->actionfunc(this, globalCtx);
+    Actor_MoveForward(&this->actor);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Yukabyun/func_80B43B6C.s")
+    if (!(this->actionfunc == func_80B43A94 || this->actionfunc == func_80B43BCC)) {
+        func_8002E4B4(globalCtx, &this->actor, 5.0f, 20.0f, 8.0f, 5);
+        Collider_CylinderUpdate(&this->actor, &this->collider);
+        
+        this->actor.flags |= 0x1000000;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Yukabyun/func_80B43BCC.s")
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+    Actor_SetHeight(&this->actor, 4.0f);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Yukabyun/EnYukabyun_Update.s")
+void EnYukabyun_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnYukabyun* this = THIS;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    Gfx* dispRefs[4];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Yukabyun/EnYukabyun_Draw.s")
+    Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_en_yukabyun.c", 366);
+    func_80093D18(globalCtx->state.gfxCtx);
+    gSPSegment(gfxCtx->polyOpa.p++, 0x08, SEGMENTED_TO_VIRTUAL(D_80B43F64[(u8)this->unk_152]));
```

Comment:
is the cast here needed?

Reply:
If I take away the cast I get an lb instead of a lbu on line 480 of the diff. Not sure if there's a way to remove the cast and also fix that

Reply:
is unk152 unsigned then?